transactions/brc20.ts

```diff
@@ -37,3 +60,306 @@ export const createBrc20TransferOrder = async (
+  const {
+    seedPhrase,
+    accountIndex,
+    addressUtxos,
+    tick,
+    amount,
+    revealAddress,
+    changeAddress,
+    feeRate,
+    recipientAddress,
+    network,
+  } = executeProps;
+
+  const privateKey = await getBtcPrivateKey({
+    seedPhrase,
+    index: BigInt(accountIndex),
+    network: 'Mainnet',
+  });
+
+  const esploraClient = new BitcoinEsploraApiProvider({ network });
+
+  yield ExecuteTransferProgressCodes.CreatingInscriptionOrder;
+
+  const finalRecipientUtxoValue = new BigNumber(FINAL_SATS_VALUE);
+  const { tx } = await signNonOrdinalBtcSendTransaction(
+    recipientAddress,
+    [
+      {
+        address: revealAddress,
+        status: {
+          confirmed: false,
+        },
+        txid: '0000000000000000000000000000000000000000000000000000000000000001',
+        vout: 0,
+        value: FINAL_SATS_VALUE,
+      },
+    ],
+    accountIndex,
+    seedPhrase,
+    'Mainnet',
+    new BigNumber(1),
+  );
+
+  const transferFeeEstimate = tx.vsize * feeRate;
+
+  const inscriptionValue = finalRecipientUtxoValue.plus(transferFeeEstimate);
+
+  const { commitAddress, commitValue } = await xverseInscribeApi.createBrc20TransferOrder(
+    tick,
+    amount,
+    revealAddress,
+    feeRate,
+    network,
+    inscriptionValue.toNumber(),
+  );
+
+  yield ExecuteTransferProgressCodes.CreatingCommitTransaction;
+
+  const bestUtxoData = selectUtxosForSend(
+    changeAddress,
+    [{ address: revealAddress, amountSats: new BigNumber(commitValue) }],
+    addressUtxos,
+    feeRate,
+  );
+
+  if (!bestUtxoData) {
+    throw new Error('Not enough funds at selected fee rate');
+  }
+
+  const commitChainFees = bestUtxoData.fee;
+
+  const commitTransaction = await generateSignedBtcTransaction(
+    privateKey,
+    bestUtxoData.selectedUtxos,
+    new BigNumber(commitValue),
+    [
+      {
+        address: commitAddress,
+        amountSats: new BigNumber(commitValue),
+      },
+    ],
+    changeAddress,
+    new BigNumber(commitChainFees),
+    network,
+  );
+
+  yield ExecuteTransferProgressCodes.ExecutingInscriptionOrder;
+
+  const { revealTransactionId, revealUTXOVOut, revealUTXOValue } = await xverseInscribeApi.executeBrc20Order(
+    commitAddress,
+    commitTransaction.hex,
+  );
+
+  yield ExecuteTransferProgressCodes.CreatingTransferTransaction;
+
+  const transferTransaction = await signNonOrdinalBtcSendTransaction(
+    recipientAddress,
+    [
+      {
+        address: revealAddress,
+        status: {
+          confirmed: false,
+        },
+        txid: revealTransactionId,
+        vout: revealUTXOVOut,
+        value: revealUTXOValue,
+      },
+    ],
+    accountIndex,
+    seedPhrase,
+    'Mainnet',
+    new BigNumber(transferFeeEstimate),
+  );
+
+  yield ExecuteTransferProgressCodes.Finalizing;
+
+  // we sleep here to give the reveal transaction time to propagate
+  await new Promise((resolve) => setTimeout(resolve, 500));
```

Comment:
we sleep even if the function paused at the last yield for some unknown time?

Reply:
I think it makes sense, otherwise the caller would need to know to add a sleep after `yield ExecuteTransferProgressCodes.Finalizing`